src/Components/Calculator.js

```diff
@@ -1,35 +1,78 @@
 import react from 'react';
-
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends react.Component {
   constructor() {
     super();
     this.state = {
-      Calc: ['AC', '+/-', '%', '/', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+',
-      ],
+      total: '0',
+      next: null,
+      operation: null,
     };
+    this.evntHandler = this.evntHandler.bind(this);
   }
 
-  render() {
-    return (
-      <div className="container">
-        <div className="display-screen">0</div>
-        <div className="num-rows">
-          {this.state.Calc.map((item) => (
-            <div className="butn" key={item}>
-              {item}
-            </div>
-          ))}
-        </div>
-        <div className="bottom-row">
-          <div className="butn">0</div>
-          <div className="butn">.</div>
-          <div className="butn-equals">=</div>
+    evntHandler = (e) => {
+      if (!e.target.name) return;
+
+      const { next, total, operation } = calculate(this.state, e.target.name);
+
+      if (next === null && total === null) {
+        this.setState({ next, total: '0', operation });
+      } else {
+        this.setState({ next, total, operation });
+      }
+    };
+
+    render() {
+      const { total, next } = this.state;
+      const btns = [
+        'AC',
+        '+/-',
+        '%',
+        '÷',
+        '7',
+        '8',
+        '9',
+        'x',
+        '4',
+        '5',
+        '6',
+        '-',
+        '1',
+        '2',
+        '3',
+        '+',
+        '0',
+        '.',
+        '=',
+      ];
+      return (
+        <div className="container">
+          {next ? (
+            <div className="display-screen">{next}</div>
+          ) : (
+            <div className="display-screen">{total}</div>
+          )}
+          <div className="num-rows">
+            {btns.map((btnName) => (
+              <button
+                onClick={(e) => this.evntHandler(e)}
+                name={btnName}
+                type="button"
+                className={`butn ${btnName === '0' ? 'butn-zero' : ''} ${
+                  btnName === '=' ? 'butn-equals' : ''
+                }`}
+                key={btnName}
+              >
+                {btnName}
+              </button>
+            ))}
+          </div>
```

Comment:
- [ ] Awesome, smart implementation of the events 💯. You rock 🥇

Reply:
Thank you